obspy/io/y/core.py

```diff
@@ -256,10 +307,11 @@ def _read_y(filename, headonly=False, **kwargs):  # @UnusedVariable
                 # UCHAR Key[16]
                 #   Unique key that identifies this record in the database.
                 data = fh.read(next_tag)
-                parts = unpack(endian + b'd16s', data[8:])
+                parts = _unpack_with_asciiz_and_decode(
+                    endian + b'd16s', data[8:])
```

Comment:
We have a real file. How the char header fields are expected to be treated is not stated in the manual though, so I wouldn't know how to define what ultimately is expected.

Anyway, I'd merge this now, so that this fix gets into `master`. Anybody who is using this format can work on (or at least trigger) any further polishing that wasn't done back when this was first added to obspy.